test/conftest.py

```diff
@@ -0,0 +1,28 @@
+import pytest
+
+from test.pylib.report_plugin import ReportPlugin
+
+ALL_MODES = {'debug': 'Debug',
+             'release': 'RelWithDebInfo',
+             'dev': 'Dev',
+             'sanitize': 'Sanitize',
+             'coverage': 'Coverage'}
+
+def pytest_addoption(parser):
+    parser.addoption('--mode', choices=ALL_MODES.keys(), dest="mode",
+                     help="Run only tests for given build mode(s)")
+    parser.addoption('--tmpdir', action='store', default='testlog', help='''Path to temporary test data and log files. The data is
+            further segregated per build mode. Default: ./testlog.''', )
```

Comment:
This will tell pytest where to put Junit and allure results, where to put logs.
If you don't need some functionality, it doesn't mean that nobody needs it. We need to run tests against different modes: release, debug, and dev. That's why we need to differentiate them: logs, results, binaries.

Reply:
A general note to all people I review: When I say "some this unclear" it doesn't mean just that I want you to explain it to me now - it means I think that future readers will not understand it either, so a better comment, better string, better naming, or something like that is probably needed.

Your comment "If you don't need some functionality, it doesn't mean that nobody needs it." makes no sense. You are adding here options which will apply to all test suites, you can always add different options to individual test suites. If you think this option makes sense for all test suites, you can make a better effort to explain why in a comment or the option's text. 

I don't strongly mind that you add this option if a few tests need it (I was much more strongly opposed to the "mode" and "run_id" options, and it didn't help me), by I can't figure out the help string "Path to temporary test data and log files. The data is further segregated per build mode. Default: ./testlog.". I don't understand what pytest will write to this directory - files, subdirectories, or what. Is there a specific place where the test needs to put files or directories in this directory? It's very vague.

Reply:
> A general note to all people I review: When I say "some this unclear" it doesn't mean just that I want you to explain it to me now - it means I think that future readers will not understand it either, so a better comment, better string, better naming, or something like that is probably needed.

I can't agree with you. Not all that unclear to the reviewer should be in the description. For example, there was a question from a reviewer about why I'm adding empty python files. This is definitely not worth mentioning in the description. Here is the same thing, I'm working in the context of test.py that has this option and most of the developers are aware of it. 

> Your comment "If you don't need some functionality, it doesn't mean that nobody needs it." makes no sense. You are adding here options which will apply to all test suites, you can always add different options to individual test suites. If you think this option makes sense for all test suites, you can make a better effort to explain why in a comment or the option's text

My statement is makes sense: you are not using test.py, and you do not know some basic things about it, you are using your run/run.py wrappers instead. I'm adding here zero new options, I'm moving an already existing option from a bunch of files to one.

>  I don't understand what pytest will write to this directory - files, subdirectories, or what. Is there a specific place where the test needs to put files or directories in this directory? It's very vague.

It will write files (allure, junit, xml, Scylla logs), subdirectories(separate build mode, scylla instance directory, allure, xml). Yes, there are specific folders where you should put logs if you want them to be parsed, Junit/Allure reports if you want them present in Jenkins.